src/main/resources/db/changesets/20190402_files.yaml

```diff
@@ -0,0 +1,76 @@
+databaseChangeLog:
+  - changeSet:
+      id: file_tables
+      author: dd
+      changes:
+        - createTable:
+            tableName: fs_object
+            columns:
+              - column:
+                  name: object_id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: study_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+              - column:
+                  name: object_type
+                  type: char(1)
+                  constraints:
+                    nullable: false
+                  ## D for directories; F for files; N for files not yet present
+              - column:
+                  name: created_date
+                  type: timestamptz
+                  defaultValueComputed: now()
+                  constraints:
+                    nullable: false
+              - column:
+                  name: creating_flight_id
+                  type: text
+                  ## stairway flight id to prevent create race conditions
+              - column:
+                  name: path
+                  type: text
+                  constraints:
+                    nullable: false
+                    unique: true
```

Comment:
As long as we use the C-locale in CloudSQL, this should work. If not then we'll need to create a `text_pattern_ops` index. From the [docs](https://www.postgresql.org/docs/current/indexes-opclass.html):

> The operator classes text_pattern_ops, varchar_pattern_ops, and bpchar_pattern_ops support B-tree indexes on the types text, varchar, and char respectively. The difference from the default operator classes is that the values are compared strictly character by character rather than according to the locale-specific collation rules. This makes these operator classes suitable for use by queries involving pattern matching expressions (LIKE or POSIX regular expressions) when the database does not use the standard “C” locale. As an example, you might index a varchar column like this:

```sql
CREATE INDEX test_index ON test_table (col varchar_pattern_ops);
```